### ISSUE TITLE
Cannot build without messing with /usr/local

### ISSUE THREAD

**Issue by andersk** [tuzz/ipasir-sys#1]
I run NixOS and don’t have a `/usr/local`. While I could create one through various impure methods, it’s weird and unfriendly for a package to require this. I needed to make these changes to get the crate to build without it:

```udiff
--- a/build.rs
+++ b/build.rs
@@ -29,8 +29,7 @@ fn main() {
     println!("cargo:rustc-link-search={}", build.out_dir);
     println!("cargo:rustc-link-lib=static=ipasir");
 
-    println!("cargo:rustc-link-search=/usr/local/lib");
-    println!("cargo:rustc-link-lib=static=stdc++");
+    println!("cargo:rustc-link-lib=stdc++");
 }
 
 struct Build {
@@ -67,7 +66,6 @@ impl Build {
     fn configure_cadical(&self) {
         Command::new("./configure")
             .arg("-fPIC")
-            .env("CXX", "/usr/local/bin/g++")
             .current_dir(format!("{}/cadical", self.out_dir))
             .spawn().unwrap().wait().unwrap();
     }
```

Is there a reason you can’t just respect the default `g++` in the `PATH`, which should already include `/usr/local/bin` on systems that have one, and ask the user to change their own `PATH` or `CXX` if they need to override it, like they would for any other package?

**Comment by tuzz** [comment]
Hi @andersk, thanks for this.

I'm not too familiar with the C++ conventions for specifying `g++` and the path to its header files. I'd welcome a pull request from someone who knows what that should be.

I just did the simplest thing I can think of that works, but I appreciate not all *nix systems have `/usr/local`.

**Comment by tuzz** [comment]
Hi again,

I've just published version `0.3.0` which should solve these problems. The crate no longer relies on `/usr/local` and tries to infer the C++ standard library location.

Please open a new issue if you are still having trouble. Thanks
